src/journal/journald-server.c

```diff
@@ -2490,15 +2507,19 @@ static int server_load_credentials(Server *s) {
                 log_debug("Acquired forward_to_console from credential.");
         }
 
-        r = server_load_boolean_credential("journald.forward_to_wall");
-        if (r < 0)
-                log_debug_errno(r, "Failed to read credential journald.forward_to_wall, ignoring: %m");
-        else {
-                s->forward_to_wall = r;
-                log_debug("Acquired forward_to_wall from credential.");
+        r = read_credential("journald.vsock_forward_address", &data, NULL);
```

Comment:
this sounds like a really good credential to have, but i think we should generalize it. i.e. no reason to limit this to vsock. Just call socket_address_parse() and allow also AF_INET, AF_INET6, AF_UNIX sockets to be created this way. For example, we want that this is useful in containers too. Hence restricting this via the name to vsock sounds counterproductive. 

hence just name this `journald.forward_address` or so, and use the generic socket address parsers.

Reply:
There was a question around whether it is okay to do blocking writes to sockets which are not AF_UNIX/AF_VSOCK. As well as was concern about how lossy AF_INET/AF_INET6 are.

I'm very happy to support these and it would simplify the code.

Currently the implementation looks for an AF_UNIX socket only at  "/run/host/journal/socket".

Also to note the socket currently uses blocking IO.

Reply:
@sam-leonard-ct Definitely use non-blocking IO. If you get EWOULDBLOCK just don't forward the message (and maybe set some flag that the next time we can send without blocking we add an extra message saying we had to drop some messages because we'd block otherwise)

Reply:
actually i would use blocking IO for this. this is a different take on the "console" concept, and that's typically synchronous. i think its fine to slow down the system if the log transport is too slow.

i mean, we don't do async IO when writing logs to serial port or VGA consoles either, even though in particular the serial port stuff is frickin slow. I see no reason to treat these socket transports differently on this.